docs/content/manual/rancher-integration/upgrade-using-rancher-ui.md

```diff
@@ -0,0 +1,142 @@
+
+AND 1 RWO and 1 RWX volumes unattached
+
+AND 1 RWO volume with 50 Gi data
+
+**When** Upgrade the Kubernetes version to latest version
+
+**Then** Upgrade should be successful
+
+AND Volumes should be healthy and data should be intact after upgrade completes
+
+AND The volumes which were attached initially should be attached and the ones which is unattached should be unattached
+
+AND Logs should not have any unexpected errors
+
+
+### 3. Upgrade multi node cluster using Rancher UI - RKE1 cluster
+
+**Given**  Multi node (1 master, 3 workers) RKE1 cluster provisioned in Rancher with K8s prior version with Longhorn installed
+
+AND few RWO and RWX volumes attached with node/pod exists
+
+AND 1 RWO and 1 RWX volumes unattached
+
+AND 1 RWO volume with 50 Gi data
+
+
+**When** Upgrade the Kubernetes version to latest version
+
+**Then** Upgrade should be successful
+
+AND Volumes should be healthy and data should be intact after upgrade completes
+
+AND The volumes which were attached initially should be attached and the ones which is unattached should be unattached
+
+AND Logs should not have any unexpected errors
+
+
+### 4. Upgrade multi node cluster using Rancher UI - K3s cluster
+
+**Given**  Multi node (1 master, 3 workers) K3s cluster provisioned in Rancher with K8s prior version with Longhorn installed
+
+AND few RWO and RWX volumes attached with node/pod exists
+
+AND 1 RWO and 1 RWX volumes unattached
+
+AND 1 RWO volume with 50 Gi data
+
+**When** Upgrade the Kubernetes version to latest version
+
+**Then** Upgrade should be successful
+
+AND Volumes should be healthy and data should be intact after upgrade completes
+
+AND The volumes which were attached initially should be attached and the ones which is unattached should be unattached
+
+AND Logs should not have any unexpected errors
+
+
+### 5. Upgrade multi node cluster using Rancher UI - K3s Imported cluster
+
+**Given**  Multi node (1 master, 3 workers) imported K3s cluster with K3s prior version with Longhorn installed
+
+AND few RWO and RWX volumes attached with node/pod exists
+
+AND 1 RWO and 1 RWX volumes unattached
+
+AND 1 RWO volume with 50 Gi data
+
+**When** Upgrade the Kubernetes version to latest version
+
+**Then** Upgrade should be successful
+
+AND Volumes should be healthy and data should be intact after upgrade completes
+
+AND The volumes which were attached initially should be attached and the ones which is unattached should be unattached
+
+AND Logs should not have any unexpected errors
+
+
+### 6. Interrupt upgrade on multi node cluster
+
+**Given**  Multi node (1 master, 3 workers) imported K3s cluster with K3s prior version with Longhorn installed
+
+AND few RWO and RWX volumes attached with node/pod exists
+
+AND 1 RWO and 1 RWX volumes unattached
+
+**When** Upgrade the Kubernetes version to latest version, while upgrade is in progress kill the kubelet process on nodes
+
+**Then** Upgrade shouldn't be successful
```

Comment:
Hi @khushboo-rancher 
    However, the upgrade was still successful because the `kubelet` process was relaunched. I executed the script on one of the nodes and kept it running to continuously kill the kubelet process. Despite this, no volume data was lost after the successful upgrade.
 
 ```
 #!/bin/bash

while true; do
  # Find the process ID of the k3s kubelet
  pid=$(ps -ef | grep kubelet | grep -v grep | awk '{print $2}')

  if [ -z "$pid" ]; then
    echo "kubelet process not found"
  else
    # Kill the k3s kubelet process
    sudo kill $pid
    echo "kubelet process killed"
  fi

  # Wait for 5 seconds before checking again
  sleep 5
done
 ```